core/src/main/java/org/elasticsearch/ElasticsearchCorruptionException.java

```diff
@@ -42,10 +42,7 @@ public ElasticsearchCorruptionException(String message) {
      * @param ex the exception cause
      */
     public ElasticsearchCorruptionException(Throwable ex) {
-        this(ex.getMessage());
-        if (ex != null) {
-            this.setStackTrace(ex.getStackTrace());
-        }
+        super(ex);
         Throwable[] suppressed = ex.getSuppressed();
```

Comment:
We should nuke all this logic after the `super` call, because now we init the exception with `ex` as root cause, so it will still keep all of its suppressed exceptions.